### PR TITLE
[FIX] fix the issue of docker image pull, add proxy for docker service

### PR DESCRIPTION
**Pull request by yejianquan**
### Description of PR

Summary:
Fixes the host unreachable issue of docker pull.

### Type of change

- [x] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Fixes the host unreachable issue of docker pull.

#### How did you do it?
Add docker proxy configuration in DUT during deploy-mg

#### How did you verify/test it?
Deploy-mg on physical testbeds and docker pull works as expected.

### CODE REVIEW

File: ansible/config_sonic_basedon_testbed.yml [hunk 1]
```diff
@@ -346,6 +346,32 @@
...
+        systemd:
+          state: restarted
+          daemon_reload: yes
+          name: docker
+
+      - name: Wait 10s for docker containers to restart
+        pause:
+          seconds: 10
+    become: true
+    when: proxy_env['https_proxy'] is defined
```
Comment (wangxin) [inline ansible/config_sonic_basedon_testbed.yml:373]: If dict `proxy_env` is not defined, will calling `proxy_env['http_proxy']` fail with something like KeyError?

Comment (yejianquan) [reply ansible/config_sonic_basedon_testbed.yml]: good point, refined

**Review by neethajohn — APPROVED** [review]

**Comment by bingwang-ms** [comment]
Hi @yejianquan ,I attempted to login to one of the agent and do a manually `docker pull`, and it can succeed without this improvement. How can we understand it?

**Comment by wangxin** [comment]
@bingwang-ms I think the issue is not about pulling docker image from agent. The issue is pulling docker image from DUT failed. Target hosts of this ansible playbook is sonic.

**Review by wangxin — APPROVED** [review]

**Comment by yejianquan** [comment]
> @bingwang-ms I think the issue is not about pulling docker image from agent. The issue is pulling docker image from DUT failed. Target hosts of this ansible playbook is sonic.

Yes, exactly.
And if you try to pull from test servers, you can reproduce the pull issue since there's no docker proxy on them.

**Comment by bingwang-ms** [comment]
> @bingwang-ms I think the issue is not about pulling docker image from agent. The issue is pulling docker image from DUT failed. Target hosts of this ansible playbook is sonic.

Oh, yes. That's right. Thanks for the clarification.

**Pull request merged**